### PR TITLE
feat: Upgrade controller-runtime to v0.19.0

### PR DESCRIPTION
**Pull request by MisterMX**
### Description of your changes

This upgrades `sigs.k8s.io/controller-runtime` to `v0.19.0`. Since this library is prone to breaking changes some adjustments are made to apply to some changed interfaces. There should be no breaking changes for programs using only crossplane-runtime.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `earthly +reviewable` to ensure this PR is ready for review.
- [x] Added or updated unit tests.
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

File: pkg/reconciler/providerconfig/reconciler.go [hunk 1]
```diff
@@ -181,7 +181,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			if err := r.client.Delete(ctx, pcu); resource.IgnoreNotFound(err) != nil {
 				log.Debug(errDeletePCU, "error", err)
 				r.record.Event(pc, event.Warning(reasonAccount, errors.Wrap(err, errDeletePCU)))
-				return reconcile.Result{RequeueAfter: shortWait}, nil //nolint:nilerr // Returning err would make us requeue instantly.
```
Comment (phisco) [inline pkg/reconciler/providerconfig/reconciler.go:184]: looks like golangci-lint still wanted this

Comment (MisterMX) [reply pkg/reconciler/providerconfig/reconciler.go]: Fixed it. Seems there was a mismatch between the CI and my local linter.

**Review by phisco — COMMENTED** [review]
overall looks good, not super opinionated on the type aliases, leaving the last call to @negz 🙏

**Review by negz — APPROVED** [review]
Sorry for the delay - I've been out sick.

**Pull request merged**